Offline/OfflinePlayer/objc/OfflinePlayer/DownloadsViewController.m

```diff
@@ -0,0 +1,1181 @@
+			[sharedManager cancelVideoDownload:self.selectedOfflineVideoToken];
+			break;
+
+		case BCOVOfflineVideoDownloadLicensePreloaded:
+		case BCOVOfflineVideoDownloadStateCancelled:
+		case BCOVOfflineVideoDownloadStateTracksCancelled:
+		case BCOVOfflineVideoDownloadStateCompleted:
+		case BCOVOfflineVideoDownloadStateTracksCompleted:
+		case BCOVOfflineVideoDownloadStateError:
+		case BCOVOfflineVideoDownloadStateTracksError:
+			break;
+	}
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    gDownloadsViewController = self;
+
+    self.tabBarController = (UITabBarController*)self.parentViewController;
+
+    self.downloadsTableView.dataSource = self;
+    self.downloadsTableView.delegate = self;
+    [self.downloadsTableView setContentInset:UIEdgeInsetsMake(0, 0, 0, 0)];
+    [self createTableFooter];
+    [self updateInfoForSelectedDownload];
+    
+    [self createPlayerView];
+
+    {
+        UILongPressGestureRecognizer *longPress = [[UILongPressGestureRecognizer alloc]
+                                                   initWithTarget:self action:@selector(handleLongPress:)];
+        longPress.minimumPressDuration = 1.0; //seconds
+        longPress.delegate = self;
+        [self.downloadsTableView addGestureRecognizer:longPress];
+    }
+
+    [self setup];
+}
+
+- (void)refresh
+{
+    [self.downloadsTableView reloadData];
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+    
+    [self freeSpaceUpdate:nil];
+    self.freeSpaceTimer = [NSTimer scheduledTimerWithTimeInterval:3.0
+                                                           target:self
+                                                         selector:@selector(freeSpaceUpdate:)
+                                                         userInfo:nil
+                                                          repeats:YES];
+    
+    [self becomeFirstResponder];
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [self.freeSpaceTimer invalidate];
+    self.freeSpaceTimer = nil;
+}
+
+// Set a number on the "Downloads" tab icon
+- (void)updateBadge
+{
+    NSArray<BCOVOfflineVideoStatus *> *statusArray = BCOVOfflineVideoManager.sharedManager.offlineVideoStatus;
+    
+    int downloadingCount = 0;
+    
+    for (BCOVOfflineVideoStatus * offlineVideoStatus in statusArray)
+    {
+        if (offlineVideoStatus.downloadState == BCOVOfflineVideoDownloadLicensePreloaded
+            || offlineVideoStatus.downloadState == BCOVOfflineVideoDownloadStateDownloading
+            || offlineVideoStatus.downloadState == BCOVOfflineVideoDownloadStateRequested)
+        {
+            downloadingCount ++;
+        }
+    }
+    
+    NSString *badgeString;
+    if (downloadingCount > 0)
+    {
+        badgeString = [NSString stringWithFormat:@"%d", downloadingCount];
+    }
+    
+    self.tabBarController.tabBar.items[1].badgeValue = badgeString;
+}
+
+- (void)freeSpaceUpdate:(NSTimer *)timer
+{
+    const double cMB = (1024.0 * 1024.0);
```

Comment:
Let's make this (1000.0 * 1000.0) and ... * 1000.0) below that it's actually Megabytes and Gigabytes that match how Apple reports file space (instead of Mebibytes and Gibibytes).

Reply:
√